modules/dxp/apps/scim/scim-rest-impl/src/main/java/com/liferay/scim/rest/internal/manager/UserManagerImpl.java

```diff
@@ -261,7 +260,7 @@ public UsersGetResponse listUsersWithGET(
 
 					return ScimUserUtil.toUser(
 						_toScimUser(
-							_userLocalService.getUser(
+							_userService.getUserById(
```

Comment:
This can throw a permission exception. We should be returning the results that the user has permission for (similar to search containers on UI).

I think you are leaning on the the search API checking VIEW permission so theoretically this should never happen. But do you know if that check is done against an index (snapshot) of permissions, or checked against the actual User records?

If it is the former, then we need this additional permission check and to avoid messing up the pagination of results, we could consider returning a placeholder object in the HTTP result (for example only include the ID property for it). I think this because it is more of a system misconfiguration error if the SCIM client credential user does not have permission to its own users.

Reply:
Let's go with forwarding and we can create a bug ticket if it turns out to be a real issue.